include/nbl/builtin/hlsl/sort/counting.hlsl

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/builtin/hlsl/sort/common.hlsl"
+#include "nbl/builtin/hlsl/workgroup/arithmetic.hlsl"
+
+#ifndef _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sort
+{
+
+template<
+    uint16_t GroupSize,
+    uint16_t KeyBucketCount,
+    typename Key,
```

Comment:
try adding a
```cpp
using key_t = decltype(declval<KeyAccessor>().get(0));
```
and using that throughout instead of `typename Key`

Reply:
Done